resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/KubernetesSparkRestServer.scala

```diff
@@ -339,17 +347,38 @@ private[spark] object KubernetesSparkRestServer {
         " is not a file, or does not exist.")
     }
     val sslOptions = if (parsedArguments.useSsl) {
-      val keyStorePassword = parsedArguments
-        .keyStorePasswordFile
-        .map(new File(_))
-        .map(Files.toString(_, Charsets.UTF_8))
       val keyPassword = parsedArguments
         .keyPasswordFile
         .map(new File(_))
         .map(Files.toString(_, Charsets.UTF_8))
+      val keyStorePassword = parsedArguments
+        .keyStorePasswordFile
+        .map(new File(_))
+        .map(Files.toString(_, Charsets.UTF_8))
+      val resolvedKeyStore = (parsedArguments.keyStoreFile, parsedArguments.keyPemFile) match {
+        case (None, Some(keyPemFile)) =>
+          parsedArguments.certPemFile match {
```

Comment:
Not a fan of matching on options. See http://blog.originate.com/blog/2014/06/15/idiomatic-scala-your-options-do-not-match/ (Spark codebase doesn't like folding on them). map(...).getOrElse(throw new SparkException).